packages/aila/src/core/quiz/fullservices/CompositeFullQuizServiceBuilder.test.ts

```diff
@@ -87,46 +90,174 @@ const shouldSkipTests = process.env.TEST_QUIZZES === "false";
       expect(quiz[0]?.distractors).toBeDefined();
       log.info(JSON.stringify(quiz, null, 2));
     });
+    it("Should work with a rag quiz generator", async () => {
+      const mockRelevantLessons: AilaRagRelevantLesson[] = [
+        { lessonPlanId: "0anVg2hmAKl2YwsPjUXL0", title: "test-title-2" },
+        { lessonPlanId: "08_VNQ-oPRwaXs7hOSHtL", title: "test-title-3" },
+        { lessonPlanId: "0bz8ZgPlNRRPb5AT5hhqO", title: "test-title-4" },
+        { lessonPlanId: "0ChBXkONXh8IOVS00iTlm", title: "test-title-5" },
+      ];
 
-    it("Should work with a schema reranker", async () => {
       const builder = new CompositeFullQuizServiceBuilder();
       const settings: QuizBuilderSettings = {
         quizRatingSchema: testRatingSchema,
         quizSelector: "simple",
-        quizReranker: "schema-reranker",
-        quizGenerators: ["basedOnRag"],
+        quizReranker: "return-first",
+        quizGenerators: ["rag"],
       };
       const service = builder.build(settings);
       const quiz = await service.createBestQuiz(
         "/starterQuiz",
         CircleTheoremLesson,
+        mockRelevantLessons,
       );
       expect(quiz).toBeDefined();
       expect(quiz.length).toBeGreaterThan(0);
       expect(quiz[0]?.question).toBeDefined();
       expect(quiz[0]?.answers).toBeDefined();
       expect(quiz[0]?.distractors).toBeDefined();
+      log.info("Quiz generated with rag and return first: ", quiz);
       log.info(JSON.stringify(quiz, null, 2));
     });
-    it("Should work with a schema reranker and a ml reranker", async () => {
+
+    it("Should work with override functionality", async () => {
+      const mockRelevantLessons: AilaRagRelevantLesson[] = [
+        { lessonPlanId: "0anVg2hmAKl2YwsPjUXL0", title: "test-title-2" },
+        { lessonPlanId: "08_VNQ-oPRwaXs7hOSHtL", title: "test-title-3" },
+        { lessonPlanId: "0bz8ZgPlNRRPb5AT5hhqO", title: "test-title-4" },
+        { lessonPlanId: "0ChBXkONXh8IOVS00iTlm", title: "test-title-5" },
+      ];
+
       const builder = new CompositeFullQuizServiceBuilder();
       const settings: QuizBuilderSettings = {
         quizRatingSchema: testRatingSchema,
         quizSelector: "simple",
-        quizReranker: "schema-reranker",
-        quizGenerators: ["basedOnRag", "ml"],
+        quizReranker: "return-first",
```

Comment:
I feel like git has got a bit confused and things you're changing parts but they're actually new tests! Makes it harder to review